### PR TITLE
Add helpers to use remote images when available

### PR DESCRIPTION
**Pull request by tukantje**
# Summary

Adds helpers to use images from external sources when available.

Screenshot 2023-03-24 at 15 04 30

# To Test

1. Open Swap
2. Click on a currency to open search
3. Search for FORTUNE
4. External results list should show the logo, as opposed to no logo symbol

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @tukantje , great job!

**Review by elena-zh — APPROVED** [review]

File: src/custom/lib/hooks/useCurrencyLogoURIs/useCurrencyLogoURIsMod.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import XDaiLogo from 'assets/cow-swap/xdai.png'
 import { ADDRESS_IMAGE_OVERRIDE } from 'constants/tokens'
 import { NATIVE_CURRENCY_BUY_ADDRESS } from 'constants/index'
 import uriToHttp from 'lib/utils/uriToHttp'
+import { useTokenLogo } from '@src/custom/components/SearchModal/CurrencySearch/useTokenSearch'
```
Comment (anxolin) [inline src/custom/lib/hooks/useCurrencyLogoURIs/useCurrencyLogoURIsMod.ts:12]: Coul this be a code smell to depend in a global hook on this `SearchModal/CurrencySearch` hook?

File: src/custom/components/SearchModal/CurrencySearch/useTokenSearch.ts [hunk 2]
```diff
@@ -49,6 +49,18 @@ const tokensData = loadable(
   })
 )
 
+export function useTokenLogo(currency?: Currency | null): string | void {
+  const tokens = useAtomValue(useMemo(() => tokensData, [currency]))
+
+  if (!!currency && currency.isToken && tokens.state === 'hasData' && tokens.data) {
+    const token = tokens.data.data.find(
+      (token) => token.chainId === currency.chainId && token.address.toLowerCase() === currency.address.toLowerCase()
```
Comment (anxolin) [inline src/custom/components/SearchModal/CurrencySearch/useTokenSearch.ts:57]: why not saving `currency.address.toLowerCase()` in a const so u don't need to do that on every iteration of the loop

**Comment by tukantje** [comment]
Closing this one down, in favour of https://github.com/cowprotocol/cowswap/pull/2179 as per Slack discussion earlier (link in the PR mentioned).

**Pull request closed without merging**